planet/src/main/webapp/js/itinerary.js

```diff
@@ -21,47 +49,133 @@ function renderStartingLocation() {
     }
 }
 
-async function renderEvents() {
-    const eventsResponse = await fetch('/update-event');
-    const events = await eventsResponse.json();
+// Add an event to the firebase realtime database
+// Todo: input validation, success/failure callbacks
+async function addEvent() {
+    const eventName = document.getElementById('add-event-name').value;
+    const eventAddress = document.getElementById('add-event-address').value;
+    const eventDuration = document.getElementById('add-event-duration').value;
+    const userId = firebase.auth().currentUser.uid; 
+    
+    // Set every event's opening hours to 8am - 5pm for now
+    const openingTime = TimeRange.getTimeInMinutes(8,0);
+    const closingTime = TimeRange.getTimeInMinutes(17,0);
+
+    const eventListRef = database.ref('users/' + userId + '/currentList');
+    // Get the order number by counting existing events
+    const snap = await eventListRef.once('value');
```

Comment:
Is 'snap' short for snapshot?

maybe a more descriptive variable name would be useful, like eventListSnapshot

Reply:
You're right. Since I used snap quite a lot in this file and in my next PR too, can I address this problem altogether in my next PR?

Reply:
each PR needs to leave the main branch in a minimum level of healthiness. In this case, this PR should be modified to use descriptive variable names.

Reply:
Ah okay, fixed!